module4/exercises/03b_ping_pong_easier.cpp

```diff
@@ -20,34 +21,74 @@ class PingPong {
 
     void ping() {
         thread_local int reps = 0;
+        cout << "Ready to ping" << endl;
         while (reps < repetitions_ and play_)
         {
             unique_lock<mutex> l(m_);
             // TODO: wait to be used here + printing, reps incrementation, chainging turn to pong
+            auto myTurn = [&] {return isPingTurn_ ;};//&& play_;};
+            opponentsTurn_.wait(l, myTurn);
+            cout << "ping..." << reps << endl;
+            reps++;
+            isPingTurn_ = false;
+            opponentsTurn_.notify_all();
             this_thread::sleep_for(500ms);
         }
         if (reps >= repetitions_)
         {
             // TOOD: only print message here
+            unique_lock<mutex> l(m_);
+            auto myTurn = [&] {return isPingTurn_ ;};//&& play_;};
+            opponentsTurn_.wait(l, myTurn);
+            cout << "Ping Repetitions Reached" << endl;
+            isPingTurn_ = false;
+            opponentsTurn_.notify_all();
         }
     }
 
     void pong() {
         thread_local int reps = 0;
+        //cout << "Ready to pong" << endl;
         while (reps < repetitions_ and play_)
```

Comment:
probably `and play_` part is problematic. When stop thread set it to false, then you may not have the last pong.